core/src/main/java/org/openstack4j/openstack/internal/BaseOpenStackService.java

```diff
@@ -33,6 +33,15 @@
 
     ServiceType serviceType = ServiceType.IDENTITY;
     Function<String, String> endpointFunc;
+    
+    private static ThreadLocal<String> reqIdContainer = new ThreadLocal<String>();
+    
+    public static final String X_OPENSTACK_REQUEST_ID = "x-openstack-request-id";
```

Comment:
Can these be part of https://github.com/ContainX/openstack4j/blob/master/core/src/main/java/org/openstack4j/core/transport/ClientConstants.java class?